plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/internal/Registry.java

```diff
@@ -13,7 +13,7 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-package org.eclipse.glsp.api.registry;
```

Comment:
This interface is (indirectly) exposed by all the public Registries. While it's not necessary to make it non-internal (As it's not supposed to be used directly), I wonder if making it internal is really a good idea. Isn't there a risk of changing this interface (which should be safe, since it's internal), and then breaking the API for all non-internal registries? (ActionRegistry, etc). 

Again, there's only one such occurrence in Ecore GLSP, and it's related to the no-longer necessary workaround that provides a custom (Ecore)DIOperationHandlerRegistry, so maybe we don't expect registries to be customized anyway. But it feels weird to have public/exposed methods declared in an internal interface.

Reply:
You're right, it doesn't really make sense since they are sort-of public API anyway.